### PR TITLE
WIP: BERT training for MLPerf

### PR DESCRIPTION
**Pull request by Eliulm**
This is a work in progress for the BERT training bounty.

TODOs:

- [x] Functionality to download dataset (verify etc.)
- [x] Non-random dataset preprocessing steps
- [x]  Dataloader
- [x] Training loop
- [ ] Add model checkpoint functionality + model save
- [x] Distributed model training (Data parallel)
- [ ] Performance optimizations for <24h

### CODE REVIEW

**Comment by wozeparrot** [comment]
Can look at my preprocessing steps here: https://github.com/wozeparrot/tinygrad/blob/bert_train/extra/datasets/wikipedia.py

You don't have to do the cleanup/download straight from wikipedia because they provide the post cleaned up data here: https://drive.google.com/drive/u/0/folders/1cywmDnAsrP5-2vsr8GDc6QUc7VWe-M3v

**Comment by Eliulm** [comment]
I see, thanks for the hint!

**Comment by geohot** [comment]
Bounty locked, good luck!

**Comment by geohot** [comment]
Unlocking bounty, over 1 week has gone by without update.

**Comment by geohot** [comment]
New rule on mlperf training bounties. in order to lock the bounty, you must post weights that you trained in tinygrad that meet the accuracy requirement. Then we can look at speed and mergablity, but until there's weights, no lock.
